spring-core/src/main/java/org/springframework/core/io/AbstractFileResolvingResource.java

```diff
@@ -298,6 +301,39 @@ protected void customizeConnection(HttpURLConnection con) throws IOException {
 		con.setRequestMethod("HEAD");
 	}
 
+	/**
+	 * This implementation returns the contents of a file as a string using the
+	 * system default Charset. Provided the resource exists and the context has
+	 * access to it, the contents will be returned as a single string with line
+	 * feed characters retained.
+	 * @return the contents of the requested file as a {@code String}.
+	 * @throws FileNotFoundException in the event the file path is invalid.
+	 * @throws IOException if the file can not be read or cannot be serialzied.
+	 */
+	@Override
+	public String getContentAsString() throws IOException {
+
+		if( !exists() ) {
+			throw new FileNotFoundException(getDescription() + " cannot be found.");
+		}
+		if ( !isReadable() ) {
+			throw new IOException(getDescription() + " cannot be opened for reading.");
+		}
+		return new String(Files.readAllBytes(Paths.get(getFile().getAbsolutePath())), Charset.defaultCharset());
+	}
+
+	@Override
+	public String getContentAsString(Charset encoding) throws IOException {
```

Comment:
I would suggest renaming 'encoding' to 'charset' to avoid confusion. Technically speaking, charset has different meaning than 'encoding'. I double checked naming convention in JDK and found either 'charset' or 'cs' is used. For string type parameter, 'charsetName' is used.

Reply:
I've changed the param name to `charset`

Reply:
Let me know if there's anything else that can be improved.

Reply:
no further concern from my side